sos/report/plugins/mellanox_firmware.py

```diff
@@ -1,132 +1,481 @@
+
+        result = self.exec_cmd("mst status")
+
+        return (
+            result.get("status") == 0 and
+            "MST PCI configuration module loaded" in result.get("output", "")
+        )
+
+    def detect_devices(self):
+        """
+        Detect Mellanox devices in the system.
+
+        Returns:
+            List of unique PCI addresses or /dev/mst device paths.
+        """
+        devices = []
+        filtered = []
+
+        if self.tool_type == "mstflint" or not self.get_mst_status():
+            result = self.exec_cmd("lspci -D -d 15b3::0200")
+            pci_pattern = re.compile(
+                r"^[\da-fA-F]{1,4}:[\da-fA-F]{1,2}:[\da-fA-F]{1,2}\.[0-7]$"
+            )
+
+            for line in result.get("output", "").splitlines():
+                if line and pci_pattern.match(line.split()[0]):
+                    devices.append(line.split()[0])
+
+            for dev in sorted(devices):
+                prefix = dev.rsplit(":", 1)[0]
+
+                # Only keep one device per PCI root prefix
+                if not any(f.startswith(prefix) for f in filtered):
+                    filtered.append(dev)
+
+        else:
+            devices = [d for d in glob.glob("/dev/mst/*") if "cable" not in d]
+
+            for dev in sorted(devices):
+                prefix = dev.rsplit(".", 1)[0]
+
+                # Only keep one device per MST root prefix
+                if not any(f.startswith(prefix) for f in filtered):
+                    filtered.append(dev)
+
+        return filtered
 
     def check_enabled(self):
         """
-        Checks if this plugin should be executed at all.
-        We will only enable the plugin if there is a
-        Mellanox Technologies network adapter
+        Determine whether this plugin should be enabled.
+
+        Returns:
+            True if Mellanox devices are present in the system, False
+            otherwise.
         """
-        lspci = self.exec_cmd("lspci -D -d 15b3::")
-        return lspci['status'] == 0 and self.MLNX_STRING in lspci['output']
+        try:
+            lspci = self.exec_cmd("lspci -D -d 15b3::0200")
 
-    def collect(self):
-        if not self.get_option('allow_system_changes'):
-            self._log_info("Skipping mst/mlx cable commands as system changes"
-                           "would be made. Use --allow-system-changes to"
-                           "enable this collection.")
-            return
+            return (
+                lspci.get("status") == 0 and
+                self.MLNX_STRING in lspci.get("output", "")
+            )
 
-        # Run only if mft package is installed.
-        # flint is available from the mft package.
-        cout = self.exec_cmd('flint --version')
-        if cout['status'] != 0:
-            return
+        except Exception:
+            return False
 
-        cout = self.collect_cmd_output('mst start')
-        if cout['status'] != 0:
-            return
+    @property
+    def timeout(self):
```

Comment:
need to check if it override global configuration, i don't think that t his is the case, you can probably set 2 sec here and run the plugin to see if it complete in 2 sec or not